From 45fe2c8676035e3709780043293adcead81be2b5 Mon Sep 17 00:00:00 2001
From: Laurent Barbe <laurent@ksperis.com>
Date: Wed, 10 Apr 2013 17:47:46 -0500
Subject: [PATCH 398/938] rbd: revalidate_disk upon rbd resize
Patch-mainline: 3.10-rc2
References: fate#312983

If rbd disk is open and rbd resize is done, new size is not
visible by filesystem.  Like is done in virtio-blk and dm driver,
revalidate_disk() permits to update the bd_inode size.

Signed-off-by: Laurent Barbe <laurent@ksperis.com>
Reviewed-by: Alex Elder <elder@inktank.com>
Acked-by: Danny Al-Gaaf <dalgaaf@suse.de>

---
 drivers/block/rbd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/block/rbd.c b/drivers/block/rbd.c
index 88b2a5b..eebf5c0 100644
--- a/drivers/block/rbd.c
+++ b/drivers/block/rbd.c
@@ -2781,6 +2781,7 @@ static int rbd_dev_refresh(struct rbd_device *rbd_dev, u64 *hver)
 	else
 		ret = rbd_dev_v2_refresh(rbd_dev, hver);
 	mutex_unlock(&ctl_mutex);
+	revalidate_disk(rbd_dev->disk);
 
 	return ret;
 }
-- 
1.8.3

